src/v/cluster/service.cc

```diff
@@ -744,6 +747,24 @@ ss::future<partition_state_reply> service::get_partition_state(
     });
 }
 
+ss::future<controller_last_applied_reply> service::get_controller_last_applied(
+  controller_last_applied_request&&, rpc::streaming_context&) {
+    return ss::with_scheduling_group(get_scheduling_group(), [this]() {
+        return ss::smp::submit_to(0, [this]() {
+            if (!_controller->is_raft0_leader()) {
+                return ss::make_ready_future<controller_last_applied_reply>(
+                  controller_last_applied_reply{
+                    .result = errc::not_leader_controller});
+            }
+            return _controller->get_last_applied_offset().then(
```

Comment:
do we need a linearizable barrier here before we return committed offset? Theoretically a stale follower (making this request) could hit a stale leader (broker that thinks it is a leader but is not).

Reply:
Yeah that is a good point, then it could be the case that after all of this waiting is done, a stale response is still returned